types/mempool/priority_nonce.go

```diff
@@ -111,25 +142,26 @@ func PriorityNonceWithTxReplacement(txReplacementRule func(op, np int64, oTx, nT
 // <0: disabled, `Insert` is a no-op
 // 0: unlimited
 // >0: maximum number of transactions allowed
-func PriorityNonceWithMaxTx(maxTx int) PriorityNonceMempoolOption {
-	return func(mp *PriorityNonceMempool) {
+func PriorityNonceWithMaxTx[C comparable](maxTx int) PriorityNonceMempoolOption[C] {
+	return func(mp *PriorityNonceMempool[C]) {
 		mp.maxTx = maxTx
 	}
 }
 
 // DefaultPriorityMempool returns a priorityNonceMempool with no options.
-func DefaultPriorityMempool() Mempool {
-	return NewPriorityMempool()
+func DefaultPriorityMempool[C comparable](txPriority TxPriority[C]) Mempool {
+	return NewPriorityMempool(txPriority)
 }
 
 // NewPriorityMempool returns the SDK's default mempool implementation which
 // returns txs in a partial order by 2 dimensions; priority, and sender-nonce.
-func NewPriorityMempool(opts ...PriorityNonceMempoolOption) *PriorityNonceMempool {
-	mp := &PriorityNonceMempool{
-		priorityIndex:  skiplist.New(skiplist.LessThanFunc(txMetaLess)),
-		priorityCounts: make(map[int64]int),
+func NewPriorityMempool[C comparable](txPriority TxPriority[C], opts ...PriorityNonceMempoolOption[C]) *PriorityNonceMempool[C] {
```

Comment:
nit: If we're going to API break anyway, I sort of think we should collapse these Options into a single Options struct kind of like in [this PR](https://github.com/cosmos/cosmos-sdk/pull/15380/files#diff-8f48d95e93f7f56444ef27ebcb327b8069404fadc7e42025061ec23390b453bcR25-R59).  My reasoning is that it's easier to read and the use of a Generic type in the options chains just seems out of place.